panic in updateReconstructedDevicePaths in standalone mode

### What happened?

Must be a result of https://github.com/kubernetes/kubernetes/pull/115268/

@jsafrane

When running kubelet in standalone mode got this panic:

```
E0313 20:39:03.702548 1312451 runtime.go:79] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 294 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic({0x3a186c0?, 0x684a380})
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:75 +0x99
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x47ce9e?})
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:49 +0x75
panic({0x3a186c0, 0x684a380})
	/usr/local/google/home/skanzhelev/src/k8s.io/kubernetes/_output/local/.gimme/versions/go1.20.2.linux.amd64/src/runtime/panic.go:884 +0x213
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).updateReconstructedDevicePaths(0xc00098e360)
	pkg/kubelet/volumemanager/reconciler/reconstruct_new.go:183 +0x8a
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).reconcileNew(0xc00098e360)
	pkg/kubelet/volumemanager/reconciler/reconciler_new.go:60 +0x99
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:154 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x42f56a5?, {0x4867b20, 0xc0010d1770}, 0x1, 0xc000298060)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:155 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x5f5e100, 0x0, 0x1e?, 0x0?)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:132 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:89
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).runNew(0xc00098e360, 0x42ef7f7?)
	pkg/kubelet/volumemanager/reconciler/reconciler_new.go:30 +0x9d
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).Run(0xc0009c3b00?, 0xc000298060?)
	pkg/kubelet/volumemanager/reconciler/reconciler_common.go:150 +0x55
created by k8s.io/kubernetes/pkg/kubelet/volumemanager.(*volumeManager).Run
	pkg/kubelet/volumemanager/volume_manager.go:285 +0x29d
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0x33632ea]

goroutine 294 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0x47ce9e?})
	vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:56 +0xd7
panic({0x3a186c0, 0x684a380})
	/usr/local/google/home/skanzhelev/src/k8s.io/kubernetes/_output/local/.gimme/versions/go1.20.2.linux.amd64/src/runtime/panic.go:884 +0x213
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).updateReconstructedDevicePaths(0xc00098e360)
	pkg/kubelet/volumemanager/reconciler/reconstruct_new.go:183 +0x8a
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).reconcileNew(0xc00098e360)
	pkg/kubelet/volumemanager/reconciler/reconciler_new.go:60 +0x99
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:154 +0x3e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x42f56a5?, {0x4867b20, 0xc0010d1770}, 0x1, 0xc000298060)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:155 +0xb6
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x5f5e100, 0x0, 0x1e?, 0x0?)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:132 +0x89
k8s.io/apimachinery/pkg/util/wait.Until(...)
	vendor/k8s.io/apimachinery/pkg/util/wait/backoff.go:89
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).runNew(0xc00098e360, 0x42ef7f7?)
	pkg/kubelet/volumemanager/reconciler/reconciler_new.go:30 +0x9d
k8s.io/kubernetes/pkg/kubelet/volumemanager/reconciler.(*reconciler).Run(0xc0009c3b00?, 0xc000298060?)
	pkg/kubelet/volumemanager/reconciler/reconciler_common.go:150 +0x55
created by k8s.io/kubernetes/pkg/kubelet/volumemanager.(*volumeManager).Run
	pkg/kubelet/volumemanager/volume_manager.go:285 +0x29d
```

### What did you expect to happen?

No panic

### How can we reproduce it (as minimally and precisely as possible)?

run kubelet in standalone mode

### Anything else we need to know?

_No response_

### Kubernetes version

master branch

### Cloud provider

<details>
N/A
</details>


### OS version

<details>
N/A
</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

--- a/pkg/kubelet/volumemanager/reconciler/reconstruct_new.go
+++ b/pkg/kubelet/volumemanager/reconciler/reconstruct_new.go
@@ -180,6 +180,14 @@ func (rc *reconciler) cleanOrphanVolumes() {
 func (rc *reconciler) updateReconstructedDevicePaths() {
 	klog.V(4).InfoS("Updating reconstructed devicePaths")
 
+	if rc.kubeClient == nil {
+		// Skip reconstructing devicePath from node objects if kubelet is in standalone mode.
+		// Such kubelet is not expected to mount any attachable volume or Secrets / ConfigMap.
+		klog.V(2).InfoS("Skipped reconstruction of DevicePaths from node.status in standalone mode")
+		rc.volumesNeedDevicePath = nil
+		return
+	}
+
 	node, fetchErr := rc.kubeClient.CoreV1().Nodes().Get(context.TODO(), string(rc.nodeName), metav1.GetOptions{})
 	if fetchErr != nil {
 		// This may repeat few times per second until kubelet is able to read its own status for the first time.

